internal/peer/lifecycle/chaincode/queryapproved.go

```diff
@@ -138,54 +138,89 @@ func (a *ApprovedQuerier) Query() error {
 	}
 
 	if strings.ToLower(a.Input.OutputFormat) == "json" {
-		return printResponseAsJSON(proposalResponse, &lb.QueryApprovedChaincodeDefinitionResult{}, a.Writer)
+		return a.printResponseAsJSON(proposalResponse)
 	}
 	return a.printResponse(proposalResponse)
 }
 
+func (a *ApprovedQuerier) printResponseAsJSON(proposalResponse *pb.ProposalResponse) error {
+	if a.Input.Name != "" {
+		return printResponseAsJSON(proposalResponse, &lb.QueryApprovedChaincodeDefinitionResult{}, a.Writer)
+	}
+	return printResponseAsJSON(proposalResponse, &lb.QueryApprovedChaincodeDefinitionsResult{}, a.Writer)
+}
+
 // printResponse prints the information included in the response
 // from the server as human readable plain-text.
 func (a *ApprovedQuerier) printResponse(proposalResponse *pb.ProposalResponse) error {
-	result := &lb.QueryApprovedChaincodeDefinitionResult{}
-	err := proto.Unmarshal(proposalResponse.Response.Payload, result)
-	if err != nil {
+	if a.Input.Name != "" {
+		result := &lb.QueryApprovedChaincodeDefinitionResult{}
+		if err := proto.Unmarshal(proposalResponse.Response.Payload, result); err != nil {
+			return errors.Wrap(err, "failed to unmarshal proposal response's response payload")
+		}
+		fmt.Fprintf(a.Writer, "Approved chaincode definition for chaincode '%s' on channel '%s':\n", a.Input.Name, a.Input.ChannelID)
+		a.printSingleApprovedChaincodeDefinition(result)
+		fmt.Fprintf(a.Writer, "\n")
+		return nil
+	}
+
+	result := &lb.QueryApprovedChaincodeDefinitionsResult{}
+	if err := proto.Unmarshal(proposalResponse.Response.Payload, result); err != nil {
 		return errors.Wrap(err, "failed to unmarshal proposal response's response payload")
 	}
-	fmt.Fprintf(a.Writer, "Approved chaincode definition for chaincode '%s' on channel '%s':\n", a.Input.Name, a.Input.ChannelID)
+	fmt.Fprintf(a.Writer, "Approved chaincode definition on channel '%s':\n", a.Input.ChannelID)
```

Comment:
There could be 0 or 1 or N, therefore it would read better as a plural:

```suggestion
	fmt.Fprintf(a.Writer, "Approved chaincode definitions on channel '%s':\n", a.Input.ChannelID)
```